tests/tests_unit/test_api/test_relationships.py

```diff
@@ -0,0 +1,180 @@
+            )
+            assert isinstance(res, Relationship)
+            res = REL_API.create(
+                Relationship(
+                    external_id="1",
+                    confidence=0.5,
+                    relationship_type="belongsTo",
+                    source={"resourceId": "bbb", "resource": "Asset"},
+                    target=test,
+                )
+            )
+            assert isinstance(res, Relationship)
+            res = REL_API.create(
+                Relationship(external_id="1", confidence=0.5, relationship_type="belongsTo", source=test, target=test)
+            )
+            assert isinstance(res, Relationship)
+
+        for call in mock_rel_response.calls:
+            it = json.loads(gzip.decompress(call.request.body).decode("utf-8"))["items"][0]
+            assert isinstance(it["source"], dict)
+            assert isinstance(it["target"], dict)
+
+    def test_create_wrong_type(self, mock_rel_response):
+        with pytest.raises(ValueError):
+            res = REL_API.create(
+                Relationship(
+                    external_id="1",
+                    confidence=0.5,
+                    relationship_type="belongsTo",
+                    source=Sequence(external_id="a'"),
+                    target={"resourceId": "bbb", "resource": "Asset"},
+                )
+            )
+
+    def test_create_multiple(self, mock_rel_response):
+        rel1 = Relationship(
+            external_id="new1",
+            confidence=0.5,
+            relationship_type="flowsTo",
+            source={"resourceId": "aaa", "resource": "Asset"},
+            target={"resourceId": "bbb", "resource": "Asset"},
+        )
+        rel2 = Relationship(
+            external_id="new2",
+            confidence=0.1,
+            relationship_type="flowsTo",
+            source={"resourceId": "aaa", "resource": "Asset"},
+            target={"resourceId": "bbb", "resource": "Asset"},
+        )
+        res = REL_API.create([rel1, rel2])
+        assert isinstance(res, RelationshipList)
+        assert mock_rel_response.calls[0].response.json()["items"] == res.dump(camel_case=True)
+
+    def test_iter_single(self, mock_rel_response):
+        for rel in REL_API:
+            assert mock_rel_response.calls[0].response.json()["items"][0] == rel.dump(camel_case=True)
+
+    def test_iter_chunk(self, mock_rel_response):
+        for rel in REL_API(chunk_size=1):
+            assert mock_rel_response.calls[0].response.json()["items"] == rel.dump(camel_case=True)
+
+    def test_delete_single(self, mock_rel_response):
+        res = REL_API.delete(external_id="a")
+        assert {"items": [{"externalId": "a"}]} == jsgz_load(mock_rel_response.calls[0].request.body)
+        assert res is None
+
+    def test_delete_multiple(self, mock_rel_response):
+        res = REL_API.delete(external_id=["a"])
+        assert {"items": [{"externalId": "a"}]} == jsgz_load(mock_rel_response.calls[0].request.body)
```

Comment:
Should make this check for all the tests. Only testing that the response is correctly parsed, and not that the correct request is sent for the other tests.